jvm/benchmarks/Main.hs

```diff
@@ -116,6 +124,17 @@ benchRefs =
     ,  bench "global reference (no finalizer)" $ nfIO $ do
         _ <- newGlobalRefNonFinalized jobj
         return ()
+    -- The next two benchmarks are to be compared with one another:
+    -- The goal is to evaluate the cost of calling deleteGlobalRefNonFinalized
+    -- with or without attaching the calling thread to the JVM
+    , bench "delete global reference (not attached)" $ nfIO $ forkWait $ do
+        ref <- newGlobalRefNonFinalized jobj
+        _ <- deleteGlobalRefNonFinalized ref
+        return ()
```

Comment:
I'd expect this code to fail to run, as it is running in a thread which is not attached. It doesn't look like `forkWait` would propagate any failures so the benchmark would silently pass.